enhancements/network/egress-router.md

```diff
@@ -0,0 +1,100 @@
+---
+title: OVN Egress Router Support
+authors:
+  - "@dmellado"
+  - "@jdesousa"
+reviewers:
+  - "@squeed"
+  - "@danwinship"
+approvers:
+  - "@knobunc"
+creation-date: 2020-08-26
+last-updated: 2020-08-26
+status: provisional
+---
+
+# OVN Egress Router Support
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+Egress traffic is traffic going from OpenShift pods to external systems,
+outside of OpenShift. There are two main options for enabling egress traffic:
+Allow access to external systems from OpenShift physical node IPs or use egress
+router.
+
+In enterprise environments, egress routers are often preferred. They
+allow granular access from a specific pod, group of pods, or project to an
+external system or service. Access via node IP means all pods running on a
+given node can access external systems.
+
+An egress router is a pod that has two interfaces (eth0) and (macvlan0). Eth0
+is on the cluster network in OpenShift (internal) and macvlan0 has an IP and
+gateway from the external physical network.
+
+Pods can access the egress router service thus enabling them to access
+external services. The egress router acts as a bridge between pods and an
+external system.
+
+Traffic going out the egress router goes via node, but it will have the MAC
+address of the macvlan0 interface inside the egress router.
+
+Nowadays this is not supported when using ovn-kubernetes, mainly due to the
+lack of macvlan support, and its implementation is needed to ensure
+feature-parity with openshift-sdn.
```

Comment:
"lack of macvlan support" makes it sound like ovn-kubernetes is at fault somehow.

In openshift-sdn, the egress router was implemented by adding an openshift-sdn-specific annotation to allow a pod to request a macvlan interface. Since we have Multus now, we are not bothering to add a similar hack to ovn-kubernetes, because you can request the macvlan interface via Multus instead. (This also gives us more flexibility in the future to use interface types other than macvlan.)